web/tests/stream_data.test.js

```diff
@@ -1228,3 +1228,48 @@ test("can_access_stream_email", () => {
     page_params.is_spectator = true;
     assert.equal(stream_data.can_access_stream_email(social), false);
 });
+
+test("is_active", () => {
+    const creator_id = 1;
+    const stream_id = 102;
+
+    const test_user = {
+        user_id: creator_id,
+        email: "test@example.com",
+        full_name: "Test User",
+        is_active: true,
+    };
+
+    const sub = {
+        stream_id,
+        subscribed: false,
+        color: "blue",
+        is_muted: true,
+        invite_only: false,
+        can_remove_subscribers_group: admins_group.id,
+        date_created: 1691057093,
+        name: "Test Stream",
+        creator_id,
+    };
+
+    people.add_active_user(test_user);
+    realm.realm_can_access_all_users_group = everyone_group.id;
+    current_user.user_id = me.user_id;
+    stream_data.add_sub(sub);
+
+    // Ensure creator details can be fetched
+    const creator_details = people.get_by_user_id(test_user.user_id);
+    assert.deepStrictEqual(
+        stream_data.maybe_get_creator_details(test_user.user_id),
+        creator_details,
+    );
+
+    // Check that is_active is set correctly
+    const settings_sub = stream_settings_data.get_sub_for_settings(sub);
+    assert.strictEqual(settings_sub.is_active, true);
+
+    // Deactivate the user and re-check
+    people.deactivate(test_user);
+    const settings_sub_after_deactivation = stream_settings_data.get_sub_for_settings(sub);
+    assert.strictEqual(settings_sub_after_deactivation.is_active, false);
```

Comment:
I don't think a separate test is needed for this and you can just check this in `creator_id` test.